### ISSUE TITLE
Feature suggestion: transformations

### ISSUE THREAD

**Issue by dmarcelino** [balderdashy/waterline#772]
Hi,

It would be nice to be able to pass a function to a query which would be applied to every record after retrieval in order to transform properties/records. I don't think something like this exists currently. It's not hard do it in the callback but it would save the whole hassle of writing a loop to iterate over the retrieved records. 

A good example of this can be seen in [oriento](https://github.com/codemix/oriento#query-builder-transform-a-field):
``` javascript
db
.select('name')
.from('OUser')
.where({status: 'ACTIVE'})
.transform({
  status: function (status) {
    return status.toLowerCase();
  }
})
.limit(1)
.one()
.then(function (user) {
  console.log('user status: ', user.status); // 'active'
});
```

**Comment by mzalazar** [comment]
I need this feature right now jeje :)

**Comment by sailsbot** [comment]
Thanks for posting, @dmarcelino.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
